src/examples/shared/useCalculateScroll.ts

```diff
@@ -0,0 +1,61 @@
+import { useLayoutEffect, useState } from "react"
+import analyticsEventSections from "./example-sections/constants"
+import { majorScale } from "evergreen-ui"
+
+const clamp = (value: number) => Math.max(0, value)
+
+const isBetween = (value: number, floor: number, ceil: number) =>
+  value >= floor && value <= ceil
+
+const useCalculateScroll = (
+  offset = majorScale(17)
+): { activeTitle: string } => {
+  const [activeTitle, setActiveTitle] = useState("")
+
+  useLayoutEffect(() => {
+    const listener = () => {
+      const scroll = window.pageYOffset
+
+      const position = analyticsEventSections
+        .map((eventSection) => {
+          const { title } = eventSection
+          const element = document.getElementById(title)
+
+          if (!element) return { title, top: -1, bottom: -1 }
+
+          const rect = element.getBoundingClientRect()
+          const top = clamp(rect.top + scroll - offset)
+          const bottom = clamp(rect.bottom + scroll - offset)
+
+          return { title, top, bottom }
+        })
+        .find(({ top, bottom, title }) => {
+          const lastSectionTitles = [
```

Comment:
This could probably be simplified with `takeRight(analyticsEventSections, 2).map((section) => section.title)` 

[lodash#takeRight](https://lodash.com/docs/4.17.15#takeRight)